fe/fe-core/src/main/java/org/apache/doris/nereids/metrics/EventSwitchParser.java

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.metrics;
+
+import org.apache.doris.nereids.metrics.event.CostStateUpdateEvent;
+import org.apache.doris.nereids.metrics.event.CounterEvent;
+import org.apache.doris.nereids.metrics.event.EnforcerEvent;
+import org.apache.doris.nereids.metrics.event.FunctionCallEvent;
+import org.apache.doris.nereids.metrics.event.GroupMergeEvent;
+import org.apache.doris.nereids.metrics.event.StatsStateEvent;
+import org.apache.doris.nereids.metrics.event.TransformEvent;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * parser
+ */
+public class EventSwitchParser {
+    private static final Map<String, Class<? extends Event>> EVENT_TYPE_SET =
+            new Builder<String, Class<? extends Event>>()
+            .put("costState", CostStateUpdateEvent.class)
+            .put("counter", CounterEvent.class)
+            .put("enforcer", EnforcerEvent.class)
+            .put("functionCall", FunctionCallEvent.class)
+            .put("groupMerge", GroupMergeEvent.class)
+            .put("statsState", StatsStateEvent.class)
+            .put("transform", TransformEvent.class)
+            .build();
+
+    /**
+     * parse
+     */
+    public Set<Class<? extends Event>> parse(String eventTypeMode) {
+        List<String> strings = Arrays.stream(eventTypeMode.split("[ ,]"))
+                .map(String::trim)
+                .collect(Collectors.toList());
+        Preconditions.checkArgument(strings.size() > 0);
+        if ("all".equals(strings.get(0))) {
+            if (strings.size() == 1) {
+                return ImmutableSet.copyOf(EVENT_TYPE_SET.values());
+            }
+            Preconditions.checkArgument(strings.size() > 2 && "except".equals(strings.get(1)));
```

Comment:
u should reject invalid event switch when user set variables, should not check in here